hw4.js

```diff
@@ -0,0 +1,475 @@
+    }
+    return result;
+};
+
+/**
+ * Exercise 12
+ *
+ * Write a function that returns the 2 smallest values of an array
+ * [4,3,2,1] => [1,2]
+ *
+ * Use: .sort()
+ */
+let arr12 = [4, 3, 2, 1];
+const get2SmallestValues = (numbers) => {
+    let arrAfterSort = arr12.sort();
+    let resultArr = [];
+    resultArr.push(arrAfterSort[0], arrAfterSort[1]);
+    return resultArr;
+};
+
+/**
+ * Exercise 13
+ *
+ * Implement a function, at the input of which an object of the following form:
+ * {
+ * firstName: 'Peter',
+ * secondName: 'Vasiliev',
+ * patronymic: 'Ivanovich'
+ *}
+ * output line with the message 'Name: Petr Ivanovich Vasiliev'
+ *
+ */
+
+//it seems like a typo and here it should be "Peter" instead of "Petr"
+
+let obj = {
+    firstName: "Peter",
+    secondName: "Vasiliev",
+    patronymic: "Ivanovich",
+};
+const getFullName = (user) => {
+    let arr = Object.values(obj);
+    let arrNewOrder = [];
+    arrNewOrder.push(arr[0], arr[2], arr[1]);
+    const resultLine = arrNewOrder.join(" ");
+    return `Name: ${resultLine}`;
+};
+
+/**
+ * Exercise 14
+ *
+ * Implement a function that takes 2 arguments as input: an array of numbers and a multiplier,
+ * a returns an array of the original array, each element of which has been multiplied by a factor:
+ *
+ * [1,2,3,4], 5 => [5,10,15,20]
+ *
+ * Use: map
+ */
+const arr14 = [1, 2, 3, 4];
+const multiplyTo = (numbers, multiplier) => {
+    const result = arr14.map((number) => number * 5);
+    return result;
+};
+
+/**
+ * Exercise 15
+ *
+ * Implement a function that takes 2 arguments as input: an array and a franchise,
+ * and returns a string with the names of the heroes separated by a comma:
+ *
+ *[
+ * {name: "Batman", franchise: "DC"},
+ * {name: "Ironman", franchise: "Marvel"},
+ * {name: "Thor", franchise: "Marvel"},
+ * {name: â€œSupermanâ€, franchise: â€œDCâ€}
+ *],
+ * Marvel
+ * => Ironman, Thor
+ *
+ * Use: filter, map, join
+ */
+let arr15 = [
+    { name: "Batman", franchise: "DC" },
+    { name: "Ironman", franchise: "Marvel" },
+    { name: "Thor", franchise: "Marvel" },
+    { name: "Superman", franchise: "DC" },
+];
+const getCharacterNames = (characters, franchise) => {
+    let arrAfterFilter = arr15.filter((el) => el.franchise === "Marvel");
+    let arrAfterMap = arrAfterFilter.map((el) => {
+        return `${el.name}`;
+    });
+    const result = arrAfterMap.join(",");
+    return result;
+};
+
+// ----==== Advanced exercises (8 items) ====----
+/**
+ * Exercise 16
+ *
+ * Write a function that returns an array of the smallest row values of a two-dimensional array
+ *[
+ * [10,1,300,4],
+ * [20,2,300,400],
+ * [30,3,300,4],
+ * [40,4,300,4],
+ *]
+ * => [1,2,3,4]
+ */
+
+let arr16 = [
+    [10, 1, 300, 4],
+    [20, 2, 300, 400],
+    [30, 3, 300, 4],
+    [40, 4, 300, 4],
+];
+
+const getSmallestRow = (numbers) => {
+    let arrResult = [];
+    arr16.forEach((el) => {
+        let getMinEl = Math.min(...el);
+        arrResult.push(getMinEl);
```

Comment:
you can use `map` method instead of arrResult and do push 

```
const getSmallestRow = (numbers) => {
  return numbers.map((item) => Math.min(...item));
};